perception_eval/perception_eval/visualization/perception_visualizer3dfield.py

```diff
@@ -0,0 +1,212 @@
+        self.ax.set_yticks(field.axis_y.grid_axis * field.axis_y.plot_scale)
+
+    def plotMeshMap(self, field: PerceptionFieldXY, valuemap: np.ndarray, **kwargs) -> None:
+        x = field.mesh_center_x * field.axis_x.plot_scale
+        y = field.mesh_center_y * field.axis_y.plot_scale
+        z = valuemap
+        self.cs = self.ax.pcolormesh(x, y, z, shading="nearest", **kwargs)
+        self.cbar = self.figure.colorbar(self.cs)
+        self.cbar.set_label(self.value)
+
+    def plotScatter(self, x, y, **kwargs) -> None:
+        self.cs = self.ax.scatter(x, y, **kwargs)
+
+    def plotScatter3D(self, x, y, z, **kwargs) -> None:
+        self.ax.remove()
+        self.ax = self.figure.add_subplot(111, projection="3d")
+        self.cs = self.ax.scatter(x, y, z, **kwargs)
+        # enlarge figure size
+        self.figure.set_size_inches(14, 10)
+        self.ax.set_zlabel(self.value)
+
+
+class PerceptionFieldPlots:
+    def __init__(self, save_dir: str) -> None:
+        self.save_dir = save_dir
+        self.figures: list[PerceptionFieldPlot] = []
+
+    def add(self, figure: PerceptionFieldPlot) -> None:
+        self.figures.append(figure)
+
+    def save(self) -> None:
+        for fig in self.figures:
+            fig.figure.savefig(Path(self.save_dir, fig.name + ".png"))
+
+    def show(self) -> None:
+        plt.show()
+
+    def close(self) -> None:
+        plt.close("all")
+
+    @property
+    def last(self) -> PerceptionFieldPlot:
+        return self.figures[-1]
+
+    def plot_field_basics(self, field: PerceptionFieldXY, prefix: str, is_uncertainty: bool = False) -> None:
+        # Preprocess
+        mask_layer: np.ndarray = np.zeros(np.shape(field.num_pair), dtype=np.bool_)
+        mask_layer[field.num_pair == 0] = True
+        field.num_pair[mask_layer] = np.nan
+
+        if is_uncertainty:
+            prefix = prefix + "_uncertainty"
+
+        # Plot
+        # Number of data
+        self.add(PerceptionFieldPlot(prefix + "_" + "num", "Samples [-]"))
+        self.last.plotMeshMap(field, field.num, vmin=0)
+        self.last.setAxes(field)
+
+        # True positive rate
+        self.add(PerceptionFieldPlot(prefix + "_" + "ratio_tp", "True Positive rate [-]"))
+        self.last.plotMeshMap(field, field.ratio_tp, vmin=0, vmax=1)
+        self.last.setAxes(field)
+
+        # False positive rate
+        self.add(PerceptionFieldPlot(prefix + "_" + "ratio_fp", "False Positive rate [-]"))
+        self.last.plotMeshMap(field, field.ratio_fp, vmin=0, vmax=1)
+        self.last.setAxes(field)
+
+        if is_uncertainty == False:
+            # False negative rate
+            self.add(PerceptionFieldPlot(prefix + "_" + "ratio_fn", "False Negative rate [-]"))
+            self.last.plotMeshMap(field, field.ratio_fn, vmin=0, vmax=1)
+            self.last.setAxes(field)
+
+        # Position error
+        if field.has_any_error_data:
+            title: str = "Position error [m]"
```

Comment:
I prefer following style because it has less nested if-else. But this is just my preference.
```suggestion
            title: str = "Position uncertainty [m]" if is_uncertainty else "Position error [m]"
```

Reply:
Thanks for clean python code ;)